src/main/java/core/basesyntax/strategy/impl/SupplyOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private FruitDao fruitDao;
+
+    {
+        fruitDao = new FruitDaoImpl();
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        fruitDao = new FruitDaoImpl();
```

Comment:
Remove this line) You don't need to create a new object each time this method is called